### DISCUSSION TITLE
Keep windows on top "sticky"

### DISCUSSION THREAD

**Discussion by Vitoli-o** [rxhanson/Rectangle#289] [Ideas]
I was wondering whether it would be possible to implement a feature to always keep a window above all else (either with a shortcut or with a specific anchor).

For example, opening two Words documents side by side, with a smaller Excel spreadsheet always staying on top, regardless of whether I type on either Words document.

Also, useful for taking notes whilst watching a video.

Any chance this could be implemented?

This is available by default on both Terminal and the Simulator app, whereby you can select the option under Window to "Stay on Top".

I hope it gets implemented!

**Comment by rxhanson** [comment]
Duplicate of https://github.com/rxhanson/Rectangle/discussions/280

**Reply by Vitoli-o** [reply]
Can it be so simple as described [here](https://stackoverflow.com/questions/38711406/how-to-keep-window-always-on-the-top-with-swift) (second answer)?

I am not sure why you would need to disable SIP. Unless, you have to "wrap" the app in order to keep it always on top?

Also, [this](https://blog.rampatra.com/how-to-keep-an-apps-window-always-on-top-of-others-in-swift) mentions Apple Documentation on [NSWindow.Level](https://developer.apple.com/documentation/appkit/nswindow/level).

Perhaps, setting it to .screenSaver will allow the window to stay on top of others.

Or creating a "fake" wrapper and then setting the W and H of that particular app to match that wrapper might also work?

Just trying to do some research here... hope it helps!

**Reply by rxhanson** [reply]
Those answers are for windows that belong to your app. There is no way to obtain another application's NSWindow reference. Rectangle operates using the Accessibility API to change window size/position for windows that belong to other apps. Feel free to explore the Accessibility API and try things out to get a feel for what kinds of things you can or can't do to windows of other applications.
